python/ray/projects/scripts.py

```diff
@@ -144,6 +144,90 @@ def load_project_or_throw():
             "Project file validation failed. Please run "
             "`ray project validate` to inspect the error.")
 
+class SessionRunner(object):
+    def __init__(self, command, args, shell):
+        self.project_definition = load_project_or_throw()
+
+        # We try to parse the command here so we can fail before the cluster
+        # is started in case the command is malformed.
+        if shell:
+            self.command_to_run = command
+        else:
+            try:
+                self.command_to_run = self.project_definition.get_command_to_run(
+                    command=command, args=args)
+            except ValueError as e:
+                raise click.ClickException(e)
+
+        # Check for features we don't support right now
+        project_environment = self.project_definition.config["environment"]
+        need_docker = ("dockerfile" in project_environment
+                       or "dockerimage" in project_environment)
+        if need_docker:
+            raise click.ClickException(
+                "Docker support in session is currently not implemented. "
+                "Please file an feature request at"
+                "https://github.com/ray-project/ray/issues")
```

Comment:
I actually don't like this practice; I don't think it's particularly informative.

Reply:
(though it's not the end of the world if you leave it in)

Reply:
Yeah, I removed it